### PR TITLE
Added coverage for TelemetryView

### PR DESCRIPTION
**Pull request by WayneFerrao**
[AB#8303](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8303)

## Description

This PR adds accessibility test coverage for the TelemetryView and ContainerSummaryView components.

Note: The ContainerSummaryView component is more troublesome to test because it requires container data to properly render all its features.

### CODE REVIEW

File: packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx [hunk 1]
```diff
@@ -520,6 +521,7 @@ function FilteredTelemetryView(props: FilteredTelemetryViewProps): React.ReactEl
 		<>
 			<div style={{ display: "flex", gap: "10px" }}>
 				<Dropdown
+					aria-label="Filter Category"
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx:524]: For consistency with "Event Name Filter" below.
```suggestion
					aria-label="Category Filter"
```

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+				features: supportedFeatures,
+				devtoolsVersion: "1.0.0",
+				unsampledTelemetry: true,
+			},
+		};
+	});
+	it("TelemetryView is accessible", async () => {
+		const { container } = render(
+			<MessageRelayContext.Provider value={mockMessageRelay}>
+				<TelemetryView />{" "}
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:43]: That's not necessary, right?
```suggestion
				<TelemetryView />
```

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	it("Can tab/arrow navigate through the TelemetryView", async () => {
+		const { rerender } = render(
+			<MessageRelayContext.Provider value={mockMessageRelay}>
+				<TelemetryView />
+			</MessageRelayContext.Provider>,
+		);
+
+		const user = userEvent.setup();
+		await user.tab();
+		const telemetryConsentButton = screen.getByRole("combobox", {
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:58]: Let's use a more accurate variable name here.

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+			name: /Max Events to Display/,
+		});
+		expect(telemetryConsentButton).toHaveFocus();
+		await user.tab();
+		const refreshButton = screen.getByRole("button", { name: /Refresh Telemetry/ });
+		expect(refreshButton).toHaveFocus();
+
+		mockMessageRelay.emit("message", {
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:66]: Is emitting a message here necessary for the accessibility checks below? If so, let's add a comment to make it clear; I wouldn't have expected that to be the case.

File: packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	type DevtoolsFeatureFlags,
+	DevtoolsFeatures,
+} from "@fluidframework/devtools-core/internal";
+// Normal usage pattern for @testing-library/jest-dom
+// eslint-disable-next-line import/no-unassigned-import
+import "@testing-library/jest-dom";
+import { render } from "@testing-library/react";
+import React from "react";
+// eslint-disable-next-line import/no-internal-modules
+// import { act } from "react-dom/test-utils";
```
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx:16]: Remove commented line and unnecessary eslint disable
```suggestion
```

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx (hunk 2 above)
Comment (WayneFerrao) [reply packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:43]: Oh must have been an artifact from linting. Will fix.

**Comment by Josmithr** [comment]
> Note: The ContainerSummaryView component is more troublesome to test because it requires container data to properly render all its features.

Can that data not be easily mocked? If there is something in the API shape that makes testing difficult, we should consider ways to improve that. Not necessarily in this PR, but I'm wondering where the trouble lies.

File: packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx [hunk 6]
```diff
@@ -192,7 +192,7 @@ export function TelemetryView(): React.ReactElement {
 					)}
 				</div>
 				<div>
-					<Button onClick={handleLoadMore} size="small">
+					<Button aria-label="Refresh Telemetry" onClick={handleLoadMore} size="small">
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx:195]: For my knowledge: do you know if accessibility best practices would have us say "Refresh Telemetry Button" here? Or is that redundant given that the label is on a button element?

File: packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	type DevtoolsFeatureFlags,
+	DevtoolsFeatures,
+} from "@fluidframework/devtools-core/internal";
+// Normal usage pattern for @testing-library/jest-dom
+// eslint-disable-next-line import/no-unassigned-import
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx:11]: We may want to just update the package's eslint config to allow importing `@testing-library/jest-dom` this way.

https://github.com/import-js/eslint-plugin-import/blob/6d34c88a91a9cb7556700b7cb83c8a27731ff302/docs/rules/no-unassigned-import.md#options

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+	DevtoolsFeatures,
+} from "@fluidframework/devtools-core/internal";
+// Normal usage pattern for @testing-library/jest-dom
+// eslint-disable-next-line import/no-unassigned-import
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import React from "react";
+// eslint-disable-next-line import/no-internal-modules
+import { act } from "react-dom/test-utils";
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:17]: Is this the recommended import pattern? They don't surface test utils from the package root?

If this is the expected usage pattern, we should add an exception for this pattern to the package's eslint config directly IMO.

https://github.com/import-js/eslint-plugin-import/blob/6d34c88a91a9cb7556700b7cb83c8a27731ff302/docs/rules/no-internal-modules.md

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+import { MessageRelayContext } from "../MessageRelayContext.js";
+import { TelemetryView } from "../components/index.js";
+
+import { assertNoAccessibilityViolations, MockMessageRelay } from "./utils/index.js";
+
+describe("TelemetryView Accessibility Check", () => {
+	const supportedFeatures: DevtoolsFeatureFlags = {
+		telemetry: true,
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:26]: Should we test the accessibility when `telemetry` is false?

File: packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx (hunk 7 above)
Comment (alexvy86) [reply packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx:11]: I thought the same thing but the option is a glob for _files_, not for _packages_ 😢. We could ignore all test files but I wasn't in love with that approach.

File: packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx (hunk 6 above)
Comment (WayneFerrao) [reply packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx:195]: Screen readers will announce the element type (in this case, "button") in addition to the aria label text, so including "Button" would be redundant.

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx (hunk 9 above)
Comment (WayneFerrao) [reply packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:26]: We could, but I don't think that's necessary. Do we have use cases where telemetry is false?

File: packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx (hunk 7 above)
Comment (alexvy86) [reply packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx]: ... or maybe it works? The docs are a bit confusing in the examples.

Comment (Josmithr) [reply packages/tools/devtools/devtools-view/src/test/ContainerSummaryView.test.tsx]: Oh, huh, I thought it was globs patterns for the imports. I _think_ we leverage this elsewhere. We'd have to confirm though. If it doesn't work, please disregard the suggestion :)

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx (hunk 9 above)
Comment (Josmithr) [reply packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:23]: Yeah, if a user doesn't provide the devtools with a `DevtoolsLogger` at setup, then this flag will be false.

**Review by Josmithr — APPROVED** [review]
Left one more comment, but otherwise looks good!

File: packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx (hunk 9 above)
Comment (WayneFerrao) [reply packages/tools/devtools/devtools-view/src/test/TelemetryView.test.tsx:23]: As discussed, the TelemetryView doesn't query for supportedFeatures. If telemetry is false, the Events tab  and TelemetryView won't be rendered

**Pull request merged**